app/scripts/controllers/serviceInstance.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .controller('ServiceInstanceController', function ($scope,
+                                                     $filter,
+                                                     $routeParams,
+                                                     DataService,
+                                                     ProjectsService,
+                                                     ServiceInstancesService) {
+    $scope.alerts = {};
+    $scope.projectName = $routeParams.project;
+    $scope.serviceInstance = null;
+    $scope.serviceClass = null;
+    $scope.serviceClasses = null;
+
+    $scope.breadcrumbs = [
+      {
+        title: "Provisioned Services",
+        link: "project/" + $routeParams.project + "/browse/service-instances"
+      }
+    ];
+
+    $scope.deprovision = function() {
+      ServiceInstancesService.deprovision($scope.serviceInstance);
+    };
+
+    var watches = [];
+
+    var updateBreadcrumbs = function() {
+      if(!$scope.loaded) {
+        return;
+      }
+
+      $scope.breadcrumbs.push({
+        title: $filter('serviceInstanceDisplayName')($scope.serviceInstance, $scope.serviceClasses)
+      });
+    };
+
+    var updateServiceClassMetadata = function() {
+      if(!$scope.serviceInstance || !$scope.serviceClasses) {
+        return;
+      }
+
+      var serviceClassName = _.get($scope.serviceInstance.spec, 'serviceClassName');
+      $scope.serviceClass = _.get($scope.serviceClasses, [serviceClassName]);
+      $scope.plan = _.find(_.get($scope.serviceClass, 'plans'), {name: $scope.serviceInstance.spec.planName });
+    };
+
+    var serviceResolved = function(service, action) {
+      updateBreadcrumbs();
+      $scope.loaded = true;
+      $scope.serviceInstance = service;
+
+      if (action === "DELETED") {
+        $scope.alerts["deleted"] = {
+          type: "warning",
+          message: "This provisioned service has been deleted."
+        };
+      }
+
+      updateServiceClassMetadata();
+    };
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+        $scope.projectContext = context;
+
+        DataService
+          .get({
+            group: 'servicecatalog.k8s.io',
+            resource: 'serviceinstances'
+          }, $routeParams.instance, context, { errorNotification: false })
+          .then(function(service) {
+
```

Comment:
Call `updateBreadcrumbs` here and remove it from `serviceResolved`.

Reply:
Partially worked here... since the "crumb" display title/name is both partially dependent on ```serviceinstances``` and ```serviceclasses``` located an additional fire underneath the associated ```serviceclasses``` service call.

Reply:
Ah, I see. Then let's not set the breadcrumb until both load. Then we can call it safely from both callbacks.

Reply:
Removed the load check, instead now checking for ```serviceinstances``` and ```serviceclasses```. Similar to the check within ```updateServiceClassMetadata```.